### PR TITLE
Add space group permutations to grid.py

### PR DESCRIPTION
**Pull request by chrisrothUT**
I added space group permutations to grid.py. I also added a function which takes the translation group permutations and adds the space group on top.

### CODE REVIEW

**Comment by PhilipVinc** [comment]
Can we also get a couple of tests to check that this returns the good result in the future?

Also, how hard would it be to generalize this to arbitrary lattices, or only lattices with point symmetries such as triangular, exhagonal...

File: netket/graph/grid.py [hunk 1]
```diff
@@ -145,6 +145,80 @@ def translate(el, sites):
 
         return [translate(el, identity) for el in translation_group]
 
+    def space_group(self, identity=None) -> List[List[int]]:
```
Comment (femtobit) [inline netket/graph/grid.py:148]: `identity` is probably no longer the best term here.

If you pass `identity=[[[0, 1, 2, ..., N - 1]]]`, that is, the actual identity permutation, you would get only the space group, right?

I'm not sure what a better name is at the moment, though.

**Review by femtobit — COMMENTED** [review]
Thanks @chrisrothUT, that's very helpful already!

The space group consists of both rotations and reflections, right? It would be helpful to split `space_group` into two methods to have both types of symmetries available separately as well.

**Comment by chrisrothUT** [comment]
I will add some tests.

I think it would be hard to write something general with only access to [edges, nodes] but I'd be happy to know if I'm missing something. The easiest way would be to have common 2D lattices in Netket and implement their symmetry group by hand. I'd be willing to work on this.

File: netket/graph/grid.py (hunk 1 above)
Comment (chrisrothUT) [reply netket/graph/grid.py:148]: Yes, the space group is rotations and reflections. I think a rotation is the coordinate mapping (i,j....)->(-j,i...) which I've accomplished with two reflections. I can do this, just have to think a bit about how do handle ndim > 2.

I will think about a new name for "identity".

**Comment by chrisrothUT** [comment]
@femtobit that functionality sounds pretty useful. Frequently the ground state or low lying exciting states have some broken symmetry so it's good to be able to specify a sub-group. This project does seem pretty ambitious, perhaps we could start by implementing this for periodic 2D lattices?

**Comment by gcarleo** [comment]
since we are brainstorming here, but wouldn't it me much easier to implement translation symmetries in ```graph.Lattice```  instead of grid?

**Comment by femtobit** [comment]
@chrisrothUT Indeed - it's pretty ambitious; the good thing is, it is possible to start with what we have right now - once your PR here is finished - and at first implement `symmetries(...)` for `Grid` with support for translations + space group.

Then, additional symmetries (or finer control over existing symmetries) can be added latter in a backwards-compatible way and also to other lattices.

@gcarleo Since lattice is not a superclass of `Grid`, we'd have to do the work here anyways. So I think it's fine to start here, in particular since implementing symmetries on the grid is comparatively easy. Implementing hexagonal lattices etc. using `Lattice` in the future is an option for the future.

**Comment by femtobit** [comment]
An updated version of the space group code is now part of PR #600 (via femtobit#2), so this one can be closed.

**Pull request closed without merging**